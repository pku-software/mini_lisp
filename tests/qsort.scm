(define (my-sort l)
    (if (null? l)
        '()
        (let ((pivot (car l))
              (rest (cdr l)))
          (append (my-sort (filter (lambda (x) (< x pivot))
                                   rest))
                  (list pivot)
                  (my-sort (filter (lambda (x) (>= x pivot))
                                   rest))))))
(display (my-sort '(12 71 2 15 29 82 87 8 18 66 81 25 63 97 40 3 93 58 53 31 47)))
