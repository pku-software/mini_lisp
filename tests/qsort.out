(2 3 8 12 15 18 25 29 31 40 47 53 58 63 66 71 81 82 87 93 97)
